src/components/services/rule/rule-service.ts

```diff
@@ -33,11 +36,11 @@ export class RuleService implements IRuleService {
             catch(e){
                 //intentionally left empty as this catches errors due to resolving
             }
-            var valueMatch=(renderDescription.rule.condition.expectedValue===conditionValue);
+            var valueMatch=((<ILeafCondition>renderDescription.rule.condition).expectedValue===conditionValue);
             var effect=renderDescription.rule.effect;
             //hide
             var hide=false;
-            hide=(effect==="HIDE" && valueMatch) || (effect==="SHOW" && !valueMatch);
+            hide=(effect+''==="HIDE" && valueMatch) || (effect+''==="SHOW" && !valueMatch);
```

Comment:
the first time I tried it didn't but now it works, I changed this with the other things you pointed out